.gitlab-ci.yml

```diff
@@ -1424,3 +1424,17 @@ workflow:
       FAST_TESTS: "false"
   - variables:
       FAST_TESTS: "true"
+
+.on_invoke_tasks_changes:
+  - !reference [.except_mergequeue]
+  - changes:
+      paths:
+        - tasks/**/*
+      compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
```

Comment:
Are these supposed to run on `main`? I'd say yes, like all our other unit tests, but that is not what the rules are enforcing right now.

Reply:
Do you mean `running every time on main` versus `running only when tasks are changed`?
If yes I'm not sure the first one is mandatory (at least that's what I tried to write), if not I probably misunderstood your comment